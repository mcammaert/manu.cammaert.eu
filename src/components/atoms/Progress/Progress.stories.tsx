import React from 'react';

import { boolean, number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import Progress from './Progress';

storiesOf('Progress', module).add('Progress', () => {
  return <Progress progress={0.6} />;
});
