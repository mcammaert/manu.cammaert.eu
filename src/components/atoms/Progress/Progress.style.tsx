import styled from 'styled-components';
import { modularScale, rgba } from 'polished';

interface BarProps {
  progress: number;
}

export const Progress = styled.div`
  background-color: ${({ theme }) => rgba(theme.colors.secondary, 0.1)};
  border-radius: 3em;
`;

export const Bar = styled.div<BarProps>`
  background-color: ${({ theme }) => rgba(theme.colors.secondary, 0.8)};
  border-radius: 3em;
  width: ${({ progress }) => `${progress * 100}%`};
  height: ${modularScale(0)};
`;

export default { Progress, Bar };
