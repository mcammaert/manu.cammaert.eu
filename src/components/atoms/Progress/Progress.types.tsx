// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ProgressProps {
  className?: string;
  progress: number;
}
