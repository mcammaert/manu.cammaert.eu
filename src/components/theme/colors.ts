export const colors = {
  primary90: '#0c2d48',
  primary80: '#0b395d',
  primary70: '#0b4573',
  primary60: '#0d5189',
  primary50: '#145da0',
  primary40: '#367bb1',
  primary30: '#5c99c1',
  primary20: '#85b7d0',
  primary10: '#b1d4e0',

  secondary90: '#ca3f36',
  secondary80: '#d35239',
  secondary70: '#db633e',
  secondary60: '#e37443',
  secondary50: '#eb844a',
  secondary40: '#f79b77',
  secondary30: '#fdb5a1',
  secondary20: '#fdd0c8',
  secondary10: '#f7eceb',

  accent1: '#d45e95',
  accent2: '#b290d4',
  accent3: '#97b8e6',

  // 10 shades of grey
  black: '#1E2025',
  charcoal: '#262830',
  shadow: '#2E313A',
  lead: '#585965',
  steel: '#787C8A',
  gray: '#989DAC',
  smoke: '#BEC2C9',
  fog: '#EDEEF0',
  white: '#F6F7F9',
  snow: '#FCFCFC',

  // slate
  slateLight: '#3279AD',
  slate: '#4C6E89',
  slateDark: '#455C77',

  // teal
  teal: '#3D829C',
  tealDark: '#36768B',

  // laguna
  lagunaLight: '#4FB8F3',
  laguna: '#57A7E3',
  lagunaDark: '#3680DC',

  // warm colors
  red: '#F60C5B',
  orange: '#F86C5C',
  sunflower: '#EDC172',
  lemon: '#FFE362',

  // purples
  grape: '#374278',
  plum: '#725FA4',
  lavender: '#936FBA',
  purple: '#C163C6',

  // greens
  forest: '#50C04E',
  lawn: '#86DE6A',
  aquamarine: '#14DAB5',
  emerald: '#0DCACE',
  turquoise: '#57D5D6',

  // browns
  chocolate: '#3C322D',
  barnwood: '#5E5E5F',
};

export default colors;
