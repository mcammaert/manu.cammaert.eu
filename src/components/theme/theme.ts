import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
