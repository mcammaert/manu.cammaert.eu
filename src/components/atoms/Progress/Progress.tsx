import React from 'react';

import { ProgressProps } from './Progress.types';

import S from './Progress.style';

const Progress: React.FC<ProgressProps> = ({ progress, className }) => (
  <S.Progress className={className}>
    <S.Bar progress={progress} />
  </S.Progress>
);

export default Progress;
