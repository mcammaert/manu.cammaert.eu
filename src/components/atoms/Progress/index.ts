import * as S from './Progress.style';

export { default as Progress } from './Progress';
export * from './Progress.types';
export { S };
