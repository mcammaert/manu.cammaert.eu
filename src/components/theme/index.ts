import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
