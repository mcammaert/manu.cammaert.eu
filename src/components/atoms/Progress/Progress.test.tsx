import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import theme from 'components/theme/light';

import Progress from './Progress';

afterEach(cleanup);

describe('Progress', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Progress progress={0.6} />;
      const { container } = renderComponent({ theme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
