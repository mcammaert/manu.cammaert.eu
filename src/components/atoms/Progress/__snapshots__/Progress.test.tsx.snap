// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress default props should match the snapshot 1`] = `
.c0 {
  background-color: rgba(13,81,137,0.1);
  border-radius: 3em;
}

.c1 {
  background-color: #0d5189;
  border-radius: 3em;
  width: 60%;
  height: 1em;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;
